main/src/main/java/org/mobilitydata/gtfsvalidator/validator/StopTimeTravelSpeedValidator.java

```diff
@@ -0,0 +1,362 @@
+          return;
+        }
+      }
+    }
+  }
+
+  /**
+   * Validates travel speed between consecutive stops for all trips in a given line variant.
+   *
+   * <p>If there is a fast travel detected, then a separate notice is issued for each trip.
+   */
+  private void validateConsecutiveStops(
+      List<List<GtfsStopTime>> trips,
+      double[] distancesKm,
+      double maxSpeedInKph,
+      NoticeContainer noticeContainer) {
+    final List<GtfsStopTime> stopTimes = trips.get(0);
+
+    for (int i = 0; i < distancesKm.length; ++i) {
+      final GtfsStopTime stopTime1 = stopTimes.get(i);
+      final GtfsStopTime stopTime2 = stopTimes.get(i + 1);
+      if (!(stopTime1.hasDepartureTime() && stopTime2.hasArrivalTime())) {
+        continue;
+      }
+      final double distanceKm = distancesKm[i];
+      final double speedKph = getSpeedKphBetweenStops(distanceKm, stopTime1, stopTime2);
+      if (speedKph <= maxSpeedInKph) {
+        continue;
+      }
+      final GtfsStop stop1 = stopTable.byStopId(stopTime1.stopId());
+      final GtfsStop stop2 = stopTable.byStopId(stopTime2.stopId());
+      // Issue one notice per each trip.
+      for (List<GtfsStopTime> stopsOnTrip : trips) {
+        noticeContainer.addValidationNotice(
+            new FastTravelBetweenConsecutiveStopsNotice(
+                tripTable.byTripId(stopsOnTrip.get(0).tripId()),
+                stopsOnTrip.get(i),
+                stop1,
+                stopsOnTrip.get(i + 1),
+                stop2,
+                speedKph,
+                distanceKm));
+      }
+    }
+  }
+
+  @VisibleForTesting
+  static double getSpeedKphBetweenStops(
+      double distanceKm, GtfsStopTime stopTime1, GtfsStopTime stopTime2) {
+    return distanceKm
+        * NUM_SECONDS_PER_HOUR
+        / getTimeBetweenStops(
+            stopTime2.arrivalTime().getSecondsSinceMidnight(),
+            stopTime1.departureTime().getSecondsSinceMidnight());
+  }
+
+  private static int getTimeBetweenStops(int arrivalTime, int departureTime) {
+    int timeBetweenStops = arrivalTime - departureTime;
+    if (timeBetweenStops <= 0) {
+      // Avoid travel back in time and division by zero.
+      return NUM_SECONDS_PER_MINUTE;
+    }
+    if (arrivalTime % NUM_SECONDS_PER_MINUTE == 0 && departureTime % NUM_SECONDS_PER_MINUTE == 0) {
+      // Many scheduling systems only output times at the minute resolution,
+      // which means there can be up to 30 seconds of error in the time.  If
+      // we detect minute-resolution times, we add an extra minute to times as
+      // an error buffer.
+      timeBetweenStops += NUM_SECONDS_PER_MINUTE;
+    }
+    return timeBetweenStops;
+  }
+
+  /**
+   * The maximum distance between any two stops for which it is allowed to go over the speed limit.
+   */
+  private static final double MAX_DISTANCE_OVER_MAX_SPEED_IN_KMS = 10.0;
+
+  private static final int NUM_SECONDS_PER_MINUTE = 60;
+  private static final int NUM_SECONDS_PER_HOUR = 3600;
```

Comment:
Which manual conversion errors?

Reply:
This is effectively `60 * 60`. In projects I maintain I've tried to get away from these calculated constants as much as possible and use the built-in Java conversions instead because they protect against wasting time on silly mistakes. This specific example is obviously a super-simple case, so not a big deal to leave as is if you have a strong preference :).

Reply:
That is nice! You see, my memory keeps 3600 as a constant and not as a product `60 * 60`. I remember it from childhood this way.

I find `TimeUnit.HOURS.toSeconds(N)` convenient for `N > 1`. However, when it comes to a single hour, then I find easier to say 3600 than to rely on a library.